src-tauri/src/global/algo_store.rs

```diff
@@ -0,0 +1,107 @@
+#![allow(dead_code)]
+
+use nalgebra::DMatrix;
+use std::collections::HashMap;
+
+use crate::aux_data_structures::stoer_wagner_ds::StoerWagnerGraph;
+use crate::graph::{edge::Edge, graph_ds::Graph};
+use petgraph::graph::NodeIndex;
+
+use crate::algorithms::articulation_point::articulation_point;
+use crate::algorithms::diffcen::diffusion_centrality;
+use crate::algorithms::stoer_wagner::{recover_mincut, stoer_wagner};
+
+pub struct AlgoStore {
+    pub aps: Option<Vec<petgraph::graph::NodeIndex>>,
+    pub mincut: Option<Vec<Edge>>,
+    pub diff_cent: Option<HashMap<String, HashMap<String, f64>>>,
+    pub most_sim_t: Option<Graph>,
+    pub pred_state: Option<Graph>,
+}
+
+impl AlgoStore {
+    pub fn new() -> Self {
+        AlgoStore {
+            aps: None,
+            mincut: None,
+            diff_cent: None,
+            most_sim_t: None,
+            pred_state: None,
+        }
+    }
+
+    pub fn run_ap(&mut self, graph: &Graph) {
+        let aps = articulation_point(graph);
+        self.set_aps(aps);
+    }
+
+    pub fn run_mincut(&mut self, g: &Graph) {
+        let sw_graph = &mut StoerWagnerGraph::new(g.clone());
+        let _mincut = stoer_wagner(sw_graph).unwrap();
+        let mut nodes_string = Vec::new();
+        for node in g.get_nodes() {
+            nodes_string.push(node.name.clone());
+        }
+        let mincut_edges = recover_mincut(sw_graph, nodes_string);
+        self.set_mincut(mincut_edges);
```

Comment:
Only saving edges between mincuts here. Should we also save the weight?

Reply:
What does the weight represent? Is it an arbitrary number within the mincut problem (i.e. we're taking the max/min weighted edges), or on a set scale? (I don't quite understand mincut as a whole)

Reply:
Weight of a mincut is the sum of weights of edges between the two cuts. Basically this algorithm is dividing set of nodes into two disjoint subsets such that the sum of weights of edges between these two subsets is minimal (no other such "cut" exists with smaller weight).

Reply:
Because the edges are weighted according to their SNR and distance, their weights are already difficult to understand (i.e. as a user looking at a number on the frontend). Because the weight of a mincut is the sum of these weights, I don't imagine a user will get anything from it unless we add a legend. 

So I'm gonna vote against saving/displaying the weight, if the main info we're trying to give the user is which edges make the graph most vulnerable to disconnection.

Reply:
Fair enough, I agree. We won't have more than a few mincut edges anyway so the user can eyeball the sum if they want to.